### PR TITLE
Use the official JSON parser

### PR DESCRIPTION
**Pull request by Thog**
This remove Utf8son and JsonPrettyPrinter dependencies.

NOTE: the standard JSON parser doesn't support configurable
indentation, as a result, all the pretty printed JSON are indented with 2
spaces.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Horizon.cs [hunk 1]
```diff
@@ -519,49 +518,47 @@ public void LoadNca(Nca mainNca, Nca patchNca, Nca controlNca)
...
+            if (File.Exists(titleUpdateMetadataPath))
             {
-                using (Stream stream = File.OpenRead(Path.Combine(Device.FileSystem.GetBasePath(), "games", mainNca.Header.TitleId.ToString("x16"), "updates.json")))
+                string updatePath = JsonHelper.DeserializeFromFile<TitleUpdateMetadata>(titleUpdateMetadataPath).Selected;
+
+                if (File.Exists(updatePath))
                 {
-                    IJsonFormatterResolver resolver = CompositeResolver.Create(StandardResolver.AllowPrivateSnakeCase);
-                    string updatePath = JsonSerializer.Deserialize<TitleUpdateMetadata>(stream, resolver).Selected;
+                    FileStream file = new FileStream(updatePath, FileMode.Open, FileAccess.Read);
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Horizon.cs:529]: Looks like file is not disposed, is that intentional?

Comment (Thog) [reply Ryujinx.HLE/HOS/Horizon.cs:529]: this was already like that before so I'm not sure

Comment (emmauss) [reply Ryujinx.HLE/HOS/Horizon.cs:529]: you should probably dispose it.

Comment (Xpl0itR) [reply Ryujinx.HLE/HOS/Horizon.cs:529]: Don't dispose. That's the update file, if you dispose it then it can't be read from.

**Comment by Thog** [comment]
Rebased, ready to merge.

**Pull request merged**